src/V3Os.cpp

```diff
@@ -220,23 +252,36 @@ vluint64_t V3Os::rand64(vluint64_t* statep) {
 }
 
 string V3Os::trueRandom(size_t size) {
-    string data; data.reserve(size);
+    string result(size, '\xFF');
+    char *const data = const_cast<char*>(result.data());
+    // Note: std::string.data() returns a non-const Char* from C++17 onwards.
+    // For pre-C++17, this cast is OK in practice, even though it's UB.
+#if defined(_WIN32) || defined(__MINGW32__)
+    NTSTATUS hr = BCryptGenRandom(NULL, reinterpret_cast<BYTE*>(data), size, BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+    if (!BCRYPT_SUCCESS(hr)) {
+        v3fatal("Could not acquire random data.");
+    }
+#else
     std::ifstream is ("/dev/urandom", std::ios::in | std::ios::binary);
-    char bytes[size];
-    if (!is.read(bytes, size)) {
+    if (!is.read(data, size)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)